src/compiler/binder.ts

```diff
@@ -3509,21 +3509,22 @@ namespace ts {
 
     export function isExportsOrModuleExportsOrAlias(sourceFile: SourceFile, node: Expression): boolean {
         let i = 0;
-        const q = [node];
-        while (q.length && i < 100) {
+        const q = createQueue<Expression>();
+        q.enqueue(node);

```

Comment:
To me this part of the code seems like that order wouldnt matter so always using last element so i wonder if always using last element and keeping tail thats used to add more items and sets array.length to compact array once in a while might be better.
Thoughts?

Reply:
I didn't read it carefully when I made the change - I just checked that it would behave the same after.  Reading it now, it looks like switching from a queue to a stack would change a BFS to a DFS, which may or may not be desirable.